packages/back-end/src/routers/environment/environment.controller.ts

```diff
@@ -68,21 +67,25 @@ export const putEnvironmentOrder = async (
     context.permissions.throwPermissionError();
   }
 
-  const updatedEnvs: Environment[] = [];
+  const envIndex = existingEnvs.findIndex((env) => env.id === envId);
 
-  // Loop through env ids, to get the full env object and add it to the updatedEnvs arr
-  envIds.forEach((envId) => {
-    const env = existingEnvs.find((existing) => existing.id === envId);
+  if (envIndex < 0) {
+    return res.status(400).json({
+      status: 400,
+      message: `Unable to find environment: ${envId}`,
+    });
+  }
 
-    if (!env) {
-      return res.status(400).json({
-        status: 400,
-        message: `Unable to find environment: ${envId}`,
-      });
-    }
+  const updatedEnvs = [...existingEnvs];
 
-    updatedEnvs.push(env);
-  });
+  if (newIndex < 0 || newIndex >= existingEnvs.length) {
+    return res.status(400).json({
+      status: 400,
+      message: `Invalid new index: ${newIndex}`,
+    });
+  }
+  updatedEnvs.splice(envIndex, 1);
+  updatedEnvs.splice(newIndex, 0, updatedEnvs[envIndex]);
```

Comment:
Won't this break because at this point `updatedEnvs[envIndex]` doesn't exist anymore?